src/NuGet.Clients/PackageManagement.UI/Feeds/MultiSourcePackageMetadataProvider.cs

```diff
@@ -200,12 +200,28 @@ internal class MultiSourcePackageMetadataProvider : IPackageMetadataProvider
                 tasks.Add(_localRepository.GetPackageMetadataFromLocalSourceAsync(identity, cancellationToken));
             }
 
-            var ignored = tasks
-                .Select(task => task.ContinueWith(LogError, TaskContinuationOptions.OnlyOnFaulted))
-                .ToArray();
+            var completed = new List<IPackageSearchMetadata>();
 
-            var completed = (await Task.WhenAll(tasks))
-                .Where(m => m != null);
+            foreach (var task in tasks)
+            {
+                try
+                {
+                    var result = await task;
```

Comment:
triage question - are we going from parallel to non-parallel version of this. don't make enumerate slower/worse.